ckanext/datagovuk/plugin.py

```diff
@@ -286,16 +286,14 @@ def before_create_or_update(self, context, resource):
         # If resource is an API, don't do anything special
         if resource.get("format") == "API":
             return
-        elif "upload" not in resource:
-            return
-        elif "upload" in resource and resource["upload"] == "":
+        elif resource.get("upload") is None:
             return
         elif not upload.config_exists():
             logger = logging.getLogger(__name__)
             logger.error(
                 "Required S3 config options missing. Please check if required config options exist."
             )
-            raise Exception("Required S3 config options missing")
+            raise NameError("Required S3 config options missing")
```

Comment:
Would LookupError be a better exception? - https://docs.python.org/2/library/exceptions.html#exceptions.LookupError